### ISSUE TITLE
ColumnDefinitions in controls implementing Grid cause build failure in Android

### ISSUE THREAD

**Issue by peternary** [unoplatform/uno#718]
## I'm submitting a...

 Bug report (I searched for similar issues and did not find one)
 

## Current behavior
If you create a custom control that implements Grid and include ColumnDefinitions in the XAML then Android fails to build

## Expected behavior
Android should build

## Minimal reproduction of the problem with instructions
[App6.zip](https://github.com/nventive/Uno/files/3018117/App6.zip)
Create a project, and add a control implementing Grid.  In the xaml for that control, add ColumnDefinitions.  Attempt to build for Android - build failure.

## Environment

```
Nuget Package: 

Package Version(s): 

Affected platform(s):
- [ ] iOS
- [x] Android
- [ ] WebAssembly
- [ ] Windows
- [ ] Build tasks

Visual Studio
- [x] 2017 (version: )
- [ ] 2017 Preview (version: )
- [ ] for Mac (version: )

Relevant plugins
- [ ] Resharper (version: )
```

**Comment by carldebilly** [comment]
I successfully replicate the problem inside a sample in _Uno_.

Here's what I found so far:
1. This problem is **not specific to Android**: it's happening on all platforms (Android,. iOS & Wasm)
2. Real cause: properties assigned to `this` (root) in the code generator (`XamlCodeGeneration`)
    are not assigned the same way as other controls defines in the page. Collections seems not well
    supported there.

**Comment by carldebilly** [comment]
Fixed by PR #1356
